packages/grid/x-data-grid/src/hooks/features/headerFiltering/useGridHeaderFiltering.ts

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import { GridPrivateApiCommunity } from '../../../models/api/gridApiCommunity';
+import { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridHeaderFilteringState } from '../../../models/gridHeaderFilteringModel';
+import { useGridApiMethod } from '../../utils/useGridApiMethod';
+import { GridStateInitializer } from '../../utils/useGridInitializeState';
+import { useGridLogger } from '../../utils';
+import {
+  gridColumnLookupSelector,
+  gridColumnVisibilityModelSelector,
+  gridColumnFieldsSelector,
+} from '../columns/gridColumnsSelector';
+import {
+  GridHeaderFilteringApi,
+  GridHeaderFilteringPrivateApi,
+} from '../../../models/api/gridHeaderFilteringApi';
+
+export const headerFilteringStateInitializer: GridStateInitializer = (state) => ({
+  ...state,
+  headerFiltering: { editing: null, menuOpen: null },
+});
+
+export const useGridHeaderFiltering = (
+  apiRef: React.MutableRefObject<GridPrivateApiCommunity>,
+  props: Pick<DataGridProcessedProps, 'signature'>,
+) => {
+  const logger = useGridLogger(apiRef, 'useGridHeaderFiltering');
+
+  const setHeaderFilterState = React.useCallback(
+    (headerFilterState: Partial<GridHeaderFilteringState>) => {
+      apiRef.current.setState((state) => {
+        // Safety check to avoid MIT users from using it
+        // This hook should ultimately be moved to the Pro package
```

Comment:
Is the challenge here that these methods are being used by the code in the MIT package?

Reply:
Yes, we have to keep the state (and related methods) in MIT due to tight coupling in hooks like `useGridFocus`, `useGridKeyboardNavigation`